frontend/src/component/user/Login/Login.tsx

```diff
@@ -26,9 +27,11 @@ const Login = () => {
     const query = useQueryParams();
     const resetPassword = query.get('reset') === 'true';
     const invited = query.get('invited') === 'true';
-    const redirect = query.get('redirect') || '/';
+    const redirect =
+        query.get('redirect') || getSessionStorageItem('login-redirect') || '/';
 
     if (user) {
+        setSessionStorageItem('login-redirect', undefined);
```

Comment:
Could we default to undefined so we don't need to pass the second argument here?